crypto/src/main/java/org/web3j/crypto/transaction/type/Transaction4844.java

```diff
@@ -153,17 +153,20 @@ public List<RlpType> asRlpValues(Sign.SignatureData signatureData) {
             resultTx.add(
                     RlpString.create(
                             org.web3j.utils.Bytes.trimLeadingZeroes(signatureData.getS())));
-        }
 
-        List<RlpType> result = new ArrayList<>();
-        result.add(new RlpList(resultTx));
+            List<RlpType> result = new ArrayList<>();
+            result.add(new RlpList(resultTx));
 
-        // Adding blobs, commitments, and proofs
-        result.add(new RlpList(getRlpBlobs()));
-        result.add(new RlpList(getRlpKzgCommitments()));
-        result.add(new RlpList(getRlpKzgProofs()));
+            // Adding blobs, commitments, and proofs
+            result.add(new RlpList(getRlpBlobs()));
+            result.add(new RlpList(getRlpKzgCommitments()));
+            result.add(new RlpList(getRlpKzgProofs()));
 
-        return result;
+            return result;
+        } else {
```

Comment:
There are two RLP representations of a 4844 transaction: one for block body (without sidecar) and one "pooled transaction" for txpool/networking (with sidecar): https://eips.ethereum.org/EIPS/eip-4844#networking

I don't have enough context to know if switching on the presence of a signature is appropriate for that difference.  Is it possible a client may want to pass in a signature for a pooled transaction? 
Perhaps this class needs to expose the two possible RLP representations to the clients since maybe only the clients will have the appropriate context.

Reply:
> when the whole pooled transaction was signed, the signer was decoded incorrectly.

That's a little surprising to me given what I was reading yesterday, but I'm not very close to this code so just wanted to highlight the two possible contexts.